backend/app/Main.hs

```diff
@@ -24,35 +25,38 @@ redisConfig = defaultConnectInfo
 logType :: LogType
 logType = LogStdout defaultBufSize
 
+corsMiddleware :: Application -> Application
+corsMiddleware = cors $ const $ Just policy
+  where
+    policy =
+      CorsResourcePolicy
+        { corsOrigins = Just (["http://127.0.0.1:3000"], True),
+          corsMethods = ["POST"],
+          corsRequestHeaders = ["content-type"],
+          corsExposedHeaders = Nothing,
+          corsMaxAge = Nothing,
+          corsVaryOrigin = True,
+          corsRequireOrigin = False,
+          corsIgnoreFailures = True
+        }
+
 app :: Env -> Application
-app env = cors (const $ Just CorsResourcePolicy {
-  corsOrigins = Just (["http://127.0.0.1:3000"], True),
-  corsMethods = ["POST"],
-  corsRequestHeaders = ["content-type"],
-  corsExposedHeaders = Nothing,
-  corsMaxAge = Nothing,
-  corsVaryOrigin = True,
-  corsRequireOrigin = False,
-  corsIgnoreFailures = True
-  }) . serve api . hoistServer api (nt env) $ server
+app env = corsMiddleware . serve api . hoistServer api (nt env) $ server
 
 main :: IO ()
 main = do
   createDirectoryIfMissing True . takeDirectory $ databaseFile
 
-  dbPool <- newPool $ defaultPoolConfig (open databaseFile) close 30 10
+  -- TODO Fix up debug logging
+  dbPool <- runStdoutLoggingT $ createSqlitePoolWithConfig (toText databaseFile) defaultConnectionPoolConfig
   redisPool <- connect redisConfig
   timeCache <- newTimeCache simpleTimeFormat
-  (logger, rmLogger) <- newTimedFastLogger timeCache logType
+  (logger, _) <- newTimedFastLogger timeCache logType
 
   let env = Env {dbPool, redisPool, logger}
 
-  runReaderT initializeDatabase env
+  -- TODO Disable/handle auto-migration
```

Comment:
This automatically performs database migrations if the schema changes - not good for production, but excellent for testing migrations or for the rapid dev stage.

Reply:
> is the migrateAll an auto-generated function from the Template Haskell in Types.Database? And migrateAll will either do migrations within an existing database if any tables contain non-matching schemas, or initialize a new database with new tables if no database exists yet?

All exactly correct!

> And does "migration" equate to generating a new table with no data - like it doesn't automagically migrate the data itself right?

I have no idea! There's a few defaults that would seem reasonable (for example, what if we add a new nullable field? Can it sorta "insert" that into the table, backfilling existing rows?), but I didn't dig much into the auto-migrate behavior. We'll have to experiment. For now - it's just super convenient that the table stays up to date without any additional effort - and without writing any SQL by hand!